payroll-processor-client/src/app/payroll/payroll-charts/payroll-charts.component.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  Component,
+  OnInit,
+} from '@angular/core';
+
+import { DataService } from 'src/app/data/data-service';
+import { NameValue } from 'src/app/data/state/data-model';
+
+@Component({
+  selector: 'app-payroll-charts',
+  templateUrl: './payroll-charts.component.html',
+  styleUrls: ['./payroll-charts.component.scss'],
+})
+export class PayrollChartsComponent implements OnInit {
+  departmentDistributionData: NameValue[] = [];
+  employeeDistributionData: NameValue[] = [];
+  view: [number, number] = [700, 400];
+
+  // options+
+  gradient = false;
+  showLegend = true;
+  showLabels = false;
+  isDoughnut = false;
+  legendPosition = 'below';
+
+  colorScheme = {
+    domain: ['#5AA454', '#A10A28', '#C7B42C', '#4287f5', '#AAAAAA'],
+  };
+
+  constructor(private dataService: DataService) {
+    this.prepData();
+  }
+
+  ngOnInit(): void {}
+
+  prepData() {
+    this.prepDepartmentDistributionData();
```

Comment:
I'd keep these in the constructor.

If you find yourself wanting to break up a large function/method into smaller ones and the bigger one then calls all the smaller ones, there's probably a 'single responsibility' violation.

ex)

```typescript
someBigMethod() {
   // do thing a
   
   // do thing b

   // ...

   // do thing z
}

// becomes

someBigMethod() {
   this.doThingA();
   this.doThingB();

   // ...

   this.doThingZ();
}
```

Encapsulating these procedural operations hides the complexity, it doesn't manage or resolve it.